app/utils.py

```diff
@@ -30,3 +35,23 @@ def redact_string(input_string: str) -> str:
         output_string = re.sub(REDACT_USER_DEFINED_PATTERN, "[REDACTED]", output_string)
 
     return output_string
+
+
+def download_and_encode_image(image_url, token):
+    headers = {
+        'Authorization': f'Bearer {token}'
+    }
+    response = requests.get(image_url, headers=headers)
+    if response.status_code != 200:
+        raise ValueError(f"Request to {image_url} failed with status code {response.status_code}")
+    else:
+        content_type = response.headers['content-type']
+        if 'image' not in content_type:
+            raise ValueError(f"Content type {content_type} is not an image")
+
+        try:
+            image = Image.open(BytesIO(response.content))
+            image_format = image.format
+        except Exception as e:
+            raise ValueError(f"Error opening image: {e}")
+        return base64.b64encode(response.content).decode('utf-8'), image_format
```

Comment:
could you extract these lines of code and add some unit tests for future dependency upgrades?